File: app/src/main/java/workshop/akbolatss/xchangesrates/repositories/NotificationRepository.java
package workshop.akbolatss.xchangesrates.repositories;

import java.util.List;

import io.reactivex.Single;
import workshop.akbolatss.xchangesrates.model.dao.GlobalNotification;

/**
 * Author: Akbolat Sadvakassov
 * Date: 20.01.2018
 */

interface NotificationRepository {

    public Single<List<GlobalNotification>> getAllNotifications();

    public Single<Boolean> initDefault();
}

File: app/src/main/java/workshop/akbolatss/xchangesrates/repositories/NotificationRepository.kt
package workshop.akbolatss.xchangesrates.repositories

import io.reactivex.Single

/**
 * Author: Akbolat Sadvakassov
 * Date: 20.01.2018
 */

internal interface NotificationRepository {

//    val allNotifications: Single<List<GlobalNotification>>

    fun initDefault(): Single<Boolean>
}
